dotCMS/src/main/java/com/dotcms/publishing/PublisherAPIImpl.java

```diff
@@ -142,4 +152,78 @@ public PublishStatus publish ( PublisherConfig config, PublishStatus status ) th
         return status;
     }
 
+    @Override
+    public void addFilterDescriptor(final FilterDescriptor filterDescriptor) {
+        this.loadedFilters.put(filterDescriptor.getKey(),filterDescriptor);
+    }
+
+    @Override
+    public List<FilterDescriptor> getFiltersDescriptorsByRole(final User user) throws DotDataException {
+        if(user.isAdmin()){
+            return new ArrayList<>(this.loadedFilters.values());
+        }
+        final List<Role> roles = APILocator.getRoleAPI().loadRolesForUser(user.getUserId(), true);
+        Logger.info(this,"User Roles: " + roles.toString());
+        final List<FilterDescriptor> filters = new ArrayList<>();
+        for(final Map.Entry<String,FilterDescriptor> filterDescriptorMap : this.loadedFilters.entrySet()){
+            final String filterRoles = filterDescriptorMap.getValue().getRoles();
+            Logger.info(PublisherAPI.class,"File: " +filterDescriptorMap.getKey() + " Roles: " + filterRoles );
+            for(final Role role : roles){
+                if(UtilMethods.isSet(role.getRoleKey()) && filterRoles.contains(role.getRoleKey())){
+                    filters.add(filterDescriptorMap.getValue());
+                }
+            }
+        }
+
+        return filters;
+    }
+
+    @Override
+    public Map<String, FilterDescriptor> getFilterDescriptorMap() {
+        return this.loadedFilters;
+    }
+
+    @Override
+    public FilterDescriptor getFilterDescriptorByKey(final String filterKey) {
+        final FilterDescriptor defaultFilter = getDefaultFilter();
+        return !UtilMethods.isSet(filterKey) ? defaultFilter : this.loadedFilters.getOrDefault(filterKey,defaultFilter);
+    }
+
+    @Override
+    public PublisherFilter createPublisherFilter(final String bundleId)
+            throws DotDataException, DotSecurityException {
+
+        final String filterKey = APILocator.getBundleAPI().getBundleById(bundleId).getFilterKey();
+        final FilterDescriptor filterDescriptor = this.getFilterDescriptorByKey(filterKey);
+
+        final PublisherFilterImpl publisherFilter = new PublisherFilterImpl((Boolean)filterDescriptor.getFilters().getOrDefault("dependencies",true),
+                (Boolean)filterDescriptor.getFilters().getOrDefault("relationships",true));
+
+        if(filterDescriptor.getFilters().containsKey("excludeClasses")){
+            List.class.cast(filterDescriptor.getFilters().get("excludeClasses")).stream().forEach(type -> publisherFilter.addTypeToExcludeClassesSet(type.toString()));
+
+        }
+
+        if(filterDescriptor.getFilters().containsKey("excludeDependencyClasses")){
```

Comment:
maybe here we can use some of Polymorphism to get this code better, something like:
```suggestion
        public interface PPFilter {
            public boolean shouldApply( final FilterDescriptor filterDescriptor);
            public void. apply(final FilterDescriptor filterDescriptor, PublisherFilterImpl publisherFilter))
        }
```

Each ifs could be a implementation of the before interface, something like:

```suggestion
        public ExcludeClassPPFilter implements PPFilter {
                    public boolean shouldApply( final FilterDescriptor filterDescriptor) {
                        return filterDescriptor.getFilters().containsKey("excludeClasses");
                    }
                    
                   public void. apply(final FilterDescriptor filterDescriptor, PublisherFilterImpl publisherFilter)){
                       List.class.cast(filterDescriptor.getFilters().get("excludeClasses")).stream().forEach(type -> publisherFilter.addTypeToExcludeClassesSet(type.toString()));
                   }
        }
```